core/src/test/java/tech/tablesaw/table/TableTransposeTest.java

```diff
@@ -0,0 +1,175 @@
+package tech.tablesaw.table;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+import static tech.tablesaw.TableAssertions.assertTableContents;
+
+import org.junit.jupiter.api.Test;
+import tech.tablesaw.api.BooleanColumn;
+import tech.tablesaw.api.DoubleColumn;
+import tech.tablesaw.api.FloatColumn;
+import tech.tablesaw.api.IntColumn;
+import tech.tablesaw.api.LongColumn;
+import tech.tablesaw.api.StringColumn;
+import tech.tablesaw.api.Table;
+import tech.tablesaw.columns.numbers.DoubleColumnType;
+
+public class TableTransposeTest {
+
+  @Test
+  void transposeEmptyTable() {
+    Table empty = Table.create("Data");
+    Table result = empty.transpose();
+    assertEquals(empty.print(), result.transpose().print());
+    assertTableContents( new Object[][]{}, result);
+  }
+
+  @Test
+  void transposeDoubles() {
+    StringColumn label = StringColumn.create("label", new String[]{ "row1", "row2", "row3"});
+    DoubleColumn value = DoubleColumn.create("value1", new double[]{ 1.0, 1.1, 1.2 });
+    DoubleColumn value2 = DoubleColumn.create("value2", new double[]{ 2.0, 2.1, 2.2});
+
+    Table testTable = Table.create("Data", label, value, value2);
+    Table result = testTable.transpose();
+
+    assertTableContents(
+        new Object[][] {
+            {"label", "row1", "row2", "row3"},
+            {"value1", 1.0, 1.1, 1.2},
+            {"value2", 2.0, 2.1, 2.2},
+        },
+        result);
+    assertEquals(testTable.name(), result.name());
+    assertEquals(testTable.print(), result.transpose().print());
+  }
+
+  @Test
+  void transposeWithMissingData() {
+    StringColumn label = StringColumn.create("label", new String[]{ "row1", "row2", "row3"});
+    DoubleColumn value = DoubleColumn.create("value1", new double[] { 1.0, DoubleColumnType.missingValueIndicator(), 1.2 });
+    DoubleColumn value2 = DoubleColumn.create("value2", new double[] { 2.0, 2.1, 2.2 });
+
+    Table testTable = Table.create("Data", label, value, value2);
+    Table result = testTable.transpose();
+
+    assertTableContents(
+        new Object[][] {
+            {"label", "row1", "row2", "row3"},
+            {"value1", 1.0, Double.NaN, 1.2},
+            {"value2", 2.0, 2.1, 2.2},
+        },
+        result);
+
+    assertEquals(testTable.print(), result.transpose().print());
+  }
+
+  @Test
+  void transposeFloats() {
+    StringColumn label = StringColumn.create("label", new String[]{ "row1", "row2", "row3"});
```

Comment:
the formatting is inconsistent between this line and the next. we have an autoformatting plugin so that you don't have to worry about formatting. if you're rebased against the latest master and run `mvn compile` once on the command line it will install a git pre-commit hook that will take care of any formatting on commit

Reply:
FYI - The auto-formatting plugin doesn't work out of the box anymore. It seems to be since this change here to switch formatters:
https://github.com/jtablesaw/tablesaw/commit/646fbd494c75ca70f33b87157acfdd0a4cdf7b68
It fails locally with this error:
[ERROR] Failed to execute goal com.cosium.code:maven-git-code-format:1.38:install-hooks (install-formatter-hook) on project tablesaw-core: java.io.IOException: Cannot run program "/Applications/IntelliJ IDEA CE.app/Contents/plugins/maven/lib/maven3/bin/mvn": error=13, Permission denied -> [Help 1]
That is why I couldn't run it. The previous one did work seamlessly. Some googling suggests that I may need to install maven separately but I haven't go to the bottom of that yet. I'm using IntelliJ with its bundled maven.

Reply:
Ah. I made that switch to help Intellij users, but looks like it's hard to find a good solution :-p  The previous one only ran if you ran `mvn`, which a lot of people never did because they just used the IDE

Reply:
In case others see this, using a separate install of Maven worked here.
 
FWIW - I'm not sure I like the fact that it runs Maven now on every commit as Maven is a bit slow. I'd probably prefer instructions for contributors for how to set-up IntelliJ's formatting to match Google's so it just happens with IntelliJ magic. I appreciate that not everyone will do that though which may create more work for reviewers. Perhaps the CI can police that though, I've seen that done on other projects.